server/routes/comparison/abstractComparisonController.ts

```diff
@@ -0,0 +1,20 @@
+import { Request } from 'express'
+import HistoricalPrisonerService from '../../services/historicalPrisonerService'
+import { PrisonerDetailDto } from '../../@types/historical-prisoner/historicalPrisonerApiTypes'
+
+export default abstract class AbstractComparisonController {
+  protected constructor(protected readonly historicalPrisonerService: HistoricalPrisonerService) {}
+
+  protected async getPrisonerDetail(req: Request): Promise<PrisonerDetailDto[]> {
+    const { shortlist } = req.session
+    const results: PrisonerDetailDto[] = []
+
+    if (shortlist) {
+      const promises = shortlist.map(prisonerNo =>
+        this.historicalPrisonerService.getPrisonerDetail(req.user.token, prisonerNo),
+      )
+      results.push(...(await Promise.all(promises)))
```

Comment:
might read better if the variable is defined as it is used rather than at the top of the function